src/core/components/layout/components/columns/styles.ts

```diff
@@ -135,23 +137,84 @@ const generateWidthCSS = (width: number | number[]) => {
 	return calculateWidth(width);
 };
 
-export const column = (width?: number | number[]) => {
-	let flex;
-	let widthCSS;
-
-	if (width == null || (Array.isArray(width) && width.length === 0)) {
-		// If no width is specified, allow the column to grow
-		flex = 1;
-		widthCSS = css``;
-	} else {
-		// If a width is specified, don't allow column to grow. Use the width property
-		flex = '0 0 auto';
-		widthCSS = generateWidthCSS(width);
+const generateSpanCSS = (span: number | number[]) => {
+	if (Array.isArray(span)) {
+		const breakpoints: Breakpoint[] = [
+			'mobile',
+			'tablet',
+			'desktop',
+			'leftCol',
+			'wide',
+		];
+
+		return span.reduce((styles, w, i) => {
+			return css`
+				${styles}
+				${from[breakpoints[i]]} {
+					${calculateSpan(w)};
+				}
+			`;
+		}, css``);
 	}
 
+	return calculateSpan(span);
+};
+
+const columnBreakpoints: Array<ColumnBreakpoint> = [
+	{ totalColumns: 4, rule: until.tablet },
+	{ totalColumns: 12, rule: between.tablet.and.leftCol },
+	{ totalColumns: 14, rule: between.leftCol.and.wide },
+	{ totalColumns: 16, rule: from.wide },
+];
+
+const calculateSpan = (span: number) => {
+	const columnBreakpointCss = columnBreakpoints.reduce(
+		(acc, cur: ColumnBreakpoint) => {
+			if (span === 0)
+				return css`
+					display: none;
+				`;
```

Comment:
I think this is taken care of in [`calculateWidth`](https://github.com/guardian/source/blob/bd5a9c0486eeddbdbf258c164f1d4c5173acf0d5/src/core/components/layout/components/columns/styles.ts#L94-L104). Are you able to confirm?

Reply:
I found that it was generating some unexpected behaviour - specifically generating some blank space in the page at lower breakpoints (see below). I assumed this was some quirk of how `display: none` and `visibility: hidden` work - do you have any idea?
![image](https://user-images.githubusercontent.com/34686302/110771732-bde89e80-8252-11eb-8221-79e58b823b78.png)

Reply:
This also highlights that these changes overall are generating absurdly long class names which is bad (especially when multiple breakpoints are being specified) -  and would probably have a performance impact if it were used for lots of elements on the page.

Reply:
I've merged [a fix](https://github.com/guardian/source/pull/772) to use `display: none` instead of `visibility: hidden`. This removes the blank space and some complexity. 

I think we're going to hit issues with the `margin-right` being added to all but the last column. For instance what do we do if the last column is hidden? We'll need to find a way of removing the `margin-right` from the last _visible_ column 🤔

Reply:
(For the sake of any PR reviewers - Simon and I added a negative margin to the container so that we didn't need to identify non-visible final columns)